niconoe/pyinaturalist
# Updates to CI config & general project config

Closes Issue #28.

Deployments
==============================
* Update Travis CI config to automatically build & deploy packages to PyPi for tagged releases
    * Note: using an encrypted token under my PyPi account for deployment (which is what Travis recommends)
* Update 'Infrastructure' section and merge into 'Contributing' section

Test Coverage
==============================
* Register pyinaturalist on coveralls: https://coveralls.io/github/niconoe/pyinaturalist
* Add coverage report using `pytest-cov`
* Send test coverage results to coveralls
* Add a shiny new coveralls badge to README

Other CI Config
==============================
* Rename `flake8` env from `style` to `lint`
* Add `black --check` to `style` env
* Relax black a bit with a max line length of 100
* Run `style` env in Travis CI
* Run `mypy` env in Travis CI
* Fix a couple type annotation issues reported by mypy (oops!)
* Update `3.8-dev` env to `3.8`
* Run `coverage`, `style`, `mypy`, and `coveralls` only once, in a separate CI job under py3.8

General Config
==============================
* Require `setuptools`
* Use `setuptools.find_packages()`
* Import existing `__version__` to specify `setuptools` version
* Move static project metadata from `setup.py` to `setup.cfg`  (which supports slightly cleaner syntax for metadata)
* Let's call the development stage 'Alpha' now instead of 'Pre-Alpha'
* Remove python 3.3 from classifiers (not tested/supported)
* Move `MOCK_RESPONSE` out of `constants.py`, which is imported by
  `__init__.py`. Otherwise, the external import (`requests`) would cause, for example,
  `python setup.py --version` to fail outside a virtualenv
